tools/common/lib.go

```diff
@@ -1338,11 +1347,34 @@ func Consume(c *cli.Context, cClient ccli.Client) {
 		}()
 	}
 
-	fmt.Fprintf(os.Stdout, "%s, %s\n", `Enqueue Time (nanoseconds)`, `Message Data`)
-
+	var headerOnce sync.Once
 	for delivery := range ch {
 		msg := delivery.GetMessage()
-		_, err = fmt.Fprintf(os.Stdout, "%v, %s\n", msg.GetEnqueueTimeUtc(), string(msg.GetPayload().GetData()))
+		if _, ok := msg.GetPayload().GetUserContext()[`topic`]; ok { // If this is a Kafka-for-Cherami message, print in a JSON blob
+			m := msg.GetPayload().GetUserContext()
+			var b []byte
+			p, _ := strconv.Atoi(m[`partition`])
+			o, _ := strconv.ParseInt(m[`offset`], 10, 63)
+			b, err = json.Marshal(kafkaMessageJSON{
+				Msg:         string(msg.GetPayload().GetData()),
+				EnqueueTime: time.Unix(0, msg.GetEnqueueTimeUtc()).Format(time.StampMilli),
+				Key:         m[`key`],
+				Topic:       m[`topic`],
+				Partition:   p,
+				Offset:      o,
+			})
+			if err != nil {
+				panic(err)
+			}
+			if err == nil {
+				_, err = fmt.Fprintf(os.Stdout, "%s\n", string(b))
+			}
+		} else {
+			headerOnce.Do(func() {
```

Comment:
We want to print that header line once, and only if we are outputting non-kafka messages. This seems like it's more idiomatic than a simple boolean, for doing the header row exactly once.